Add debug logging to the service303 thread

amarpad opened the pull request:
## Summary

Move logging to the service303 thread

Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

## Test Plan

make integ_test
validated output after running test
002005 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0045    ======================================= STATISTICS ============================================
002006 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0046                   |   Current Status|
002007 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0049    Attached UEs   |          1      |
002008 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0052    Connected UEs  |          1      |
002009 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0055    Connected eNBs |          1      |
002010 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0058    Default Bearers|          1      |
002011 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0061    S1-U Bearers   |          1      |
002012 Wed Jun 30 00:59:22 2021 7FECB9095700 DEBUG SERVIC tasks/service303/service303_mme_:0066    ======================================= STATISTICS ============================================

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

Commit:
Add stats display logic to service303 thread

Change adds state display logic to service303 thread
Also adds stats for S1u and default bearer to preserve
parity with existing behavior


Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

ardzoht commented on lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_converter.cpp:
```diff
@@ -60,6 +60,14 @@ void S1apStateConverter::state_to_proto(s1ap_state_t* state, S1apState* proto) {
     FREE_HASHTABLE_KEY_ARRAY(keys);
   }
 
+  keys                        = hashtable_ts_get_keys(&state->enbs);
```
This looks to have slipped from the other PR for protobuf state eNB sync, is this PR dependent on the other one?

amarpad replied:
Hi Alex, I am not planning to merge this change instead I will merge the sub commits, this change is just to give you an overall view of where I am going with things.

ardzoht commented on lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp:
```diff
@@ -62,7 +62,6 @@ MmeNasStateManager::~MmeNasStateManager() {
 int MmeNasStateManager::initialize_state(const mme_config_t* mme_config_p) {
   persist_state_enabled = mme_config_p->use_stateless;
   max_ue_htbl_lists_    = mme_config_p->max_ues;
-  mme_statistic_timer_  = mme_config_p->mme_statistic_timer;
```
There is a `MME_STATISTIC_TIMER` config value on mme.conf.template that this value reads from, maybe we should delete that as well? Are we defaulting the display interval to 10 secs now?

amarpad replied:
Yes, we are defaulting to 10 secs. I didn't see value in changing this timer really, unless you feel strongly. I will remove this.

ardzoht reviewed (changes requested):
Some changes appear to be slipped from the other PR #7848

uri200 commented:
Those stats are being used for troubleshooting and testing. 

Will those same stats be shown after  they are moved?

amarpad commented:
> Those stats are being used for troubleshooting and testing.
> 
> Will those same stats be shown after they are moved?

I have pasted the output after the change in the commit messages's testing done section.
The following are changes from before:
1 - We will not have changes since last stat display
2 - S1 and NAS state are reported independently so there maybe a point in time where S1 shows as connected but the UEs are not showing up or the otherway around. (stats are refreshed every 60s)

amarpad commented on lte/gateway/c/core/oai/include/service303.h:
```diff
@@ -28,10 +28,12 @@
 #define NO_BOUNDARIES 0
 #define NO_LABELS 0
 #define EPC_STATS_TIMER_MSEC 60000  // In milliseconds
+#define EPC_STATS_DISPLAY_TIMER_MSEC 60000 // In milliseconds
```
@ardzoht I can plumb this through to a config file if we realize that folks want to tweak the frequency.

ardzoht replied:
Sounds good, we can keep this as it is for now

uri200 replied:
Hey @amarpad we need this to value to be configurable. The reason is when we debug issues we need to get faster feedback and know if the UE is there or not, or if the enb is up or down. Otherwise the analysis we do (specially in terravm) becomes impossible (test is finished before stats shows up)


Can we set it back to its original value until there is a setting to change that value, please?

amarpad replied:
I think you should add that logging to the S1 thread instead of this thread as S1 is the one that has the latest value. The previous implementation was wrong which was maintaining S1 state in the mme_app thread and could cause crashes.
I don't have cycles right now as I am trying to get the Sercomm stuff working. cc @talkhasib.

uri200 replied:
missed to post this here https://github.com/magma/magma/pull/8019
Please comment there if you have any issues on the modification of that value

ardzoht reviewed (approved):
lgtm
